rebasehelper/application.py

```diff
@@ -70,15 +71,20 @@ class Application:
     rebased_patches: Dict[str, List[str]] = {}
     rebased_repo: Optional[git.Repo] = None
 
-    def __init__(self, cli_conf, execution_dir, results_dir):
-        """
-        Initialize the application
+    def __init__(self, cli_conf: Config, execution_dir: str, results_dir: str, create_logs: bool = True) -> None:
+        """Initializes the application.
+
+        Args:
+            cli_conf: Application configuration.
+            execution_dir: Working directory.
+            results_dir: Location of rebase results.
+            create_logs: Whether to create default logging file handlers.
 
-        :param cli_conf: CLI object with configuration gathered from commandline
-        :return:
         """
         results_store.clear()
 
+        self.handlers = LoggerHelper.create_file_handlers(results_dir) if create_logs else None
```

Comment:
What about `self.handlers = LoggerHelper.create_file_handlers(results_dir) if create_logs else []`? This way you won't have to use the if condition in the destructor